src/Features/CSharp/Portable/CodeRefactorings/ConvertConversionOperators/ConvertConversionOperatorsFromAsRefactoringProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.ConvertConversionOperators;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertConversionOperators
+{
+    /// <summary>
+    /// Refactor:
+    ///     var o = 1 as object;
+    ///
+    /// Into:
+    ///     var o = (object)1;
+    /// </summary>
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertConversionOperators), Shared]
+    internal partial class CSharpConvertConversionOperatorsFromAsRefactoringProvider
+        : AbstractConvertConversionOperatorsRefactoringProvider<BinaryExpressionSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpConvertConversionOperatorsFromAsRefactoringProvider()
+        {
+        }
+
+        protected override Task<ImmutableArray<BinaryExpressionSyntax>> FilterFromExpressionCandidatesAsync(
+            ImmutableArray<BinaryExpressionSyntax> asExpressions,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            asExpressions = asExpressions.WhereAsArray(
+                binaryExpression => binaryExpression is
+                {
+                    RawKind: (int)SyntaxKind.AsExpression,
+                    Right: TypeSyntax { IsMissing: false },
```

Comment:
instead of this check here, an easier way to do things is have the base type check the node for .Diagnostics.  If something is missing, that will be non-empty.

Reply:
My initial concern was, that in cases like `1 as 2` the `2` would be parsed as a `LiteralExpression` and I wanted to guard the cast to `TypeSyntax` in the CodeAction. But it seems the parser always assigns a `TypeSyntax` to `asExpression.Right`. Given that, we may drop the check entirely.

I don't know if checking the diagnostics on the base expression is a good or a bad thing. I really don't care about errors on `asExpression.Left`.

IMO it's a matter of being in line with other refactorings and how those behave in case of syntax errors. I think you can make a better decision here as I can.  Just give direction to me on what's the appropriate solution here.

Reply:
> Given that, we may drop the check entirely.

Yes please :)

> I don't know if checking the diagnostics on the base expression is a good or a bad thing. I really don't care about errors on asExpression.Left.

Well, it might be nice to prevent this from showing up if you have `(as Foo)`  or (x as)`.  It's not really a problem to at least make sure we have a syntax tree without raw parser errors in it.  Up to you.  However, if you are concerned about the 'missing' case, just putthing the diag check is totally reasonable and consistent with some other refactorings.

Reply:
this still feels like something that could pull up to the base type (using ISyntaxKinds). Want me to do this?

Reply:
That would be great. I'm done for today. I also did not resolve the merge conflict because I'm still on VS 16.8. (that's why I need to place the nullable enable all over the place: I based my work on master from early September). If you take over you may also make it ready for merge.